File: OC/内容层动画/WGOCCoreAnimation/WGOCCoreAnimation/Main/ViewController.m
//
//  ViewController.m
//  WGOCAnimationPrimary
//
//  Created by wanggang on 2019/5/21.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "ViewController.h"
#import "WGCALayerAnimationViewController.h"
#import "WGWaterWaveBtnViewController.h"
#import "WGCircleBtnAniViewController.h"
#import "WGFireAniViewController.h"
#import "WGNeonAniViewController.h"
#import "WGSnowAniViewController.h"
#import "WGScanAniViewController.h"
#import "WGBeatAniViewController.h"
#import "WGChartAniViewController.h"
#import "WGEarthAniViewController.h"
#import "WGReplicatorViewController.h"
#import "WGCoverFlowViewController.h"


@interface ViewController ()<UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSArray *dataArr;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"列表";
    
    self.dataArr = @[@"基础和关键帧动画", @"水纹动画", @"渐进圆环效果", @"大火动画", @"霓虹效果", @"大雪效果", @"扫描动画", @"跳动动画", @"CAShapeLayer动画", @"地球公转", @"跳动的音符", @"CoverFlow效果"];
    [self.view addSubview:self.tableView];
    
}

#pragma mark --<UITableViewDelegate, UITableViewDataSource>方法
-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    
    return self.dataArr.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"UITableViewCell" forIndexPath:indexPath];
    cell.textLabel.text = self.dataArr[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    switch (indexPath.row) {
        case 0:{
            
            WGCALayerAnimationViewController *vc = [[WGCALayerAnimationViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 1:{
            
            WGWaterWaveBtnViewController *vc = [[WGWaterWaveBtnViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 2:{
            
            WGCircleBtnAniViewController *vc = [[WGCircleBtnAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 3:{
            
            WGFireAniViewController *vc = [[WGFireAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 4:{
            
            WGNeonAniViewController *vc = [[WGNeonAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 5:{
            
            WGSnowAniViewController *vc = [[WGSnowAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 6:{
            
            WGScanAniViewController *vc = [[WGScanAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 7:{
            
            WGBeatAniViewController *vc = [[WGBeatAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 8:{
            
            WGChartAniViewController *vc = [[WGChartAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 9:{
            
            WGEarthAniViewController *vc = [[WGEarthAniViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 10:{
            
            WGReplicatorViewController *vc = [[WGReplicatorViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        case 11:{
            
            WGCoverFlowViewController *vc = [[WGCoverFlowViewController alloc] init];
            [self.navigationController pushViewController:vc animated:YES];
            break;
        }
        default:{
            
            break;
        }
    }
}

-(UITableView *)tableView{
    
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, Height_NavBar, SCREEN_WIDTH, SCREEN_HEIGHT-Height_NavBar) style:UITableViewStylePlain];
        _tableView.backgroundColor = [UIColor whiteColor];
        _tableView.delegate = self;
        _tableView.dataSource = self;
        [_tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"UITableViewCell"];
    }
    return _tableView;
}


@end

File: Swift/显示层初级动画/WGAnimationPrimary/WGAnimationPrimary/main/ViewController.swift
//
//  ViewController.swift
//  WGAnimationPrimary
//
//  Created by wanggang on 2019/4/18.
//  Copyright © 2019 bozhong. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let tableView = UITableView.init(frame: CGRect.init(x: 0, y: 0, width: WgWith, height: WgHeight))
    let dataArr = ["几何形状动画", "位置形状动画", "淡入淡出动画和颜色", "综合位置&形状&颜色&渐变动画"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "列表"
        
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "UITableViewCell")
        tableView.delegate = self
        tableView.dataSource = self
        view.addSubview(tableView)
    }

}

//MARK:- UIViewTableView DataSource & Delegate
// 知识点：扩展
// 扩展和 Objective-C 中的分类类似，但没有名称
// 扩展可以为一个类型添加新的功能，但是不能重写已有的功能。
// 扩展可以添加新的计算型属性，但是不可以添加存储型属性，也不可以为已有属性添加属性观察器。
extension ViewController: UITableViewDelegate, UITableViewDataSource{
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataArr.count;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "UITableViewCell", for: indexPath)
        cell.textLabel?.text = dataArr[indexPath.row]
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        switch indexPath.row {
        case 0:
            
            let vc = WGGeometryAniViewController.init()
            self.navigationController?.pushViewController(vc, animated: true)
            break
            
        case 1:
            
            let vc = WGLocationAniViewController.init()
            self.navigationController?.pushViewController(vc, animated: true)
            break
            
        case 2:
            
            let vc = WGAlphaAniViewController.init()
            self.navigationController?.pushViewController(vc, animated: true)
            break
            
        case 3:
            
            let vc = WGCombinationViewController.init()
            self.navigationController?.pushViewController(vc, animated: true)
            break
            
        default:
            break
        }
    }
}
